### ISSUE TITLE
DockerComposeContainer does not remove the network after using a version 2 compose file

### ISSUE THREAD

**Issue by haines** [testcontainers/testcontainers-java#152]
I'm not using the ambassador containers so am not affected by #146, meaning that I am successfully running with a version 2 docker-compose file. This means that each test run creates a network which is not removed at the end of the test.

Once I hit 32 networks, I ran into an error when trying to run docker-compose:
```
ERROR: failed to parse pool request for address space "LocalDefault" pool "" subpool "": could not find an available predefined network
```

We need to either remove the network ourselves, or run `docker-compose down` to clean up.

**Comment by rnorth** [comment]
Good catch, thanks! As discussed, I'll work this into the docker compose
refactoring I'm doing. I think I'd like to:

- make use of docker-compose down in normal shutdown scenarios
- leave the container reaper prepped to kill any containers on exceptional
JVM shutdown
- (possibly) set up docker-compose down in a JVM shutdown hook too, to
avoid networks slipping through the cracks
On Thu, 2 Jun 2016 at 17:25 Andrew Haines <notifications@github.com> wrote:

> I'm not using the ambassador containers so am not affected by #146
> <https://github.com/testcontainers/testcontainers-java/issues/146>,
> meaning that I am successfully running with a version 2 docker-compose
> file. This means that each test run creates a network which is not removed
> at the end of the test.
>
> Once I hit 32 networks, I ran into an error when trying to run
> docker-compose:
>
> ERROR: failed to parse pool request for address space "LocalDefault" pool "" subpool "": could not find an available predefined network
>
> We need to either remove the network ourselves, or run docker-compose down
> to clean up.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/testcontainers/testcontainers-java/issues/152>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AAIETw4EFwieJn1mmtJi-LqWitAPMzuKks5qHwP-gaJpZM4IsvMn>
> .
>
-- 
Richard

**Comment by haines** [comment]
FWIW I backported the remove network command from docker-java 3.0.0-RC5 into our test suite (I see they released 3.0.0 final on Friday night!) and (with a sprinkling of nasty reflection to access private fields in the docker client), we have it successfully dropping the default network after the test run. This is sufficient for our use case; we have no plans to use additional custom networks.

**Comment by rnorth** [comment]
Resolved in #156
